integrated_channels/sap_success_factors/utils.py

```diff
@@ -166,26 +169,74 @@ def get_serialized_data_blocks(self):
                 len(this_batch)
             )
 
+    def resolve_removed_courses(self, previous_audit_summary):
+        """
+        Ensures courses that are no longer in the catalog get properly marked as inactive.
+
+        Args:
+            previous_audit_summary (dict): The previous audit summary from the last course export.
+
+        Returns:
+            An audit summary of courses with information about their presence in the catalog and current status.
+        """
+        if self.removed_courses_resolved:
+            return {}
+
+        new_audit_summary = {}
+        new_courses = []
+
+        for course in self.courses:
+            key = course['courseID']
+            status = course['status']
+
+            # Remove the key from previous audit summary so we can process courses that are no longer present,
+            # and keep course records for all previously pushed courses and new, active courses.
+            if previous_audit_summary.pop(key, None) or status == self.STATUS_ACTIVE:
+                new_courses.append(course)
+                new_audit_summary[key] = {
+                    'in_catalog': True,
+                    'status': status,
+                }
+
+        for key, summary in previous_audit_summary.items():
+            # Add a course payload to self.courses so that courses no longer in the catalog are marked inactive.
+            if summary['status'] == self.STATUS_ACTIVE and summary['in_catalog']:
+                new_courses.append({
+                    'courseID': key,
+                    'status': self.STATUS_INACTIVE,
+                })
+
+                new_audit_summary[key] = {
+                    'in_catalog': False,
+                    'status': self.STATUS_INACTIVE,
+                }
+
+        self.courses = new_courses
+        self.removed_courses_resolved = True
+        return new_audit_summary
+
     data_transform = {
         'courseID': lambda x: x['key'],
         'providerID': lambda x: apps.get_model(
             'sap_success_factors',
             'SAPSuccessFactorsGlobalConfiguration'
         ).current().provider_id,
-        'status': lambda x: 'ACTIVE' if x['availability'] == 'Current' else 'INACTIVE',
+        'status': lambda x: (SapCourseExporter.STATUS_ACTIVE if x['availability'] == 'Current'
```

Comment:
Seems like we might be able to move these availability values into a similar enum/constant thing.